src/main/ngapp/src/app/dataservices/virtualization/view-editor/command/add-sources-command.ts

```diff
@@ -66,6 +68,15 @@ export class AddSourcesCommand extends Command {
     }
 
     this._args.set( AddSourcesCommand.addedSourcePaths, arg );
+
+    if (!id) {
+      //
+      // Generate new id for this source
+      //
+      id = AddSourcesCommand.id + AddSourcesCommand.idGenerator++;
```

Comment:
Is it possible that 2 commands could end up with the same ID? For instance, if I create one command in one sessions (say ID = 0); then come back for a second session the next day and create a new command. Could that second command also have and ID of 0?

Reply:
Yes it could so fixed it with using Date.now() instead.